cpu/esp32/startup.c

```diff
@@ -211,132 +199,47 @@ NORETURN void IRAM call_start_cpu0 (void)
-#endif
-
-    LOG_STARTUP("Switching system clocks can lead to some unreadable characters\n");
-
-    /* wait until UART is idle to avoid losing output */
-    uart_tx_wait_idle(CONFIG_CONSOLE_UART_NUM);
-
-    /* determine configured CPU clock frequency from sdk_conf.h */
-    rtc_cpu_freq_t freq;
-    switch (CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ) {
-        case 40:  freq = RTC_CPU_FREQ_XTAL; /* derived from external crystal */
-                  break;                    /* normally 40 MHz */
-        case 80:  freq = RTC_CPU_FREQ_80M;  /* derived from PLL */
-                  break;
-        case 160: freq = RTC_CPU_FREQ_160M; /* derived from PLL */
-                  break;
-        case 240: freq = RTC_CPU_FREQ_240M; /* derived from PLL */
-                  break;
-        default:  freq = RTC_CPU_FREQ_2M;   /* frequencies <= 8 MHz are
-                                               set to 2 MHz and handled later */
-    }
-
-    uint32_t freq_before = esp_clk_cpu_freq() / MHZ;
-
-    if (freq_before != CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ) {
-        rtc_cpu_freq_config_t clk_cfg;
-        rtc_clk_cpu_freq_to_config(freq, &clk_cfg);
-
-        /* set configured CPU frequency */
-        rtc_clk_cpu_freq_set_config(&clk_cfg);
-
-        /* Recalculate the ccount to make time calculation correct. */
-        uint32_t freq_after = CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ;
-        XTHAL_SET_CCOUNT( XTHAL_GET_CCOUNT() * freq_after / freq_before );
-    }
-}
-
-extern void IRAM_ATTR thread_yield_isr(void* arg);
-
 static NORETURN void IRAM system_init (void)
 {
+#if defined(MCU_ESP32)
     /* enable cached read from flash */
     Cache_Read_Enable(PRO_CPU_NUM);
+#endif
 
     /* initialize the ISR stack for usage measurements */
     thread_isr_stack_init();
 
-    /* initialize clocks (CPU_CLK, APB_CLK, SLOW and FAST) */
-    system_clk_init();
-
-    /* disable clocks of peripherals that are not needed at startup */
-    esp_perip_clk_init();
-    g_startup_time = esp_rtc_get_time_us();
-
-    /* set configured console UART baudrate */
-    const int uart_clk_freq = rtc_clk_apb_freq_get();
-    uart_tx_wait_idle(CONFIG_CONSOLE_UART_NUM);
-    uart_div_modify(CONFIG_CONSOLE_UART_NUM,
-                    (uart_clk_freq << 4) / STDIO_UART_BAUDRATE);
-
-    /* initialize system call tables of ESP32 rom and newlib */
+    /* initialize system call tables of ESP32x rom and newlib */
     syscalls_init();
 
     /* install exception handlers */
     init_exceptions();
 
-    /* clear interrupt matrix */
-    intr_matrix_clear();
-
     /* systemwide UART initialization */
     extern void uart_system_init (void);
     uart_system_init();
 
-    /* Disable the hold flag of all RTC GPIO pins */
-    RTCCNTL.hold_force.val = 0;
-
     /* set log levels for SDK library outputs */
     extern void esp_log_level_set(const char* tag, esp_log_level_t level);
     esp_log_level_set("wifi", LOG_DEBUG);
+    esp_log_level_set("gpio", LOG_DEBUG);
```

Comment:
Won't this increase verbosity by a lot? Or is this not what I think it is?

Reply:
It doesn't increase verbosity, it just sets the maximum level allowed for the gpio component.